group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -822,21 +834,60 @@ public CompletableFuture<DeleteGroupsResponseData.DeletableGroupResultCollection
         });
 
         groupsByTopicPartition.forEach((topicPartition, groupList) -> {
-            CompletableFuture<DeleteGroupsResponseData.DeletableGroupResultCollection> future =
-                runtime.scheduleWriteOperation(
-                    "delete-groups",
-                    topicPartition,
-                    Duration.ofMillis(config.offsetCommitTimeoutMs()),
-                    coordinator -> coordinator.deleteGroups(context, groupList)
-                ).exceptionally(exception -> handleOperationException(
-                    "delete-groups",
-                    groupList,
-                    exception,
-                    (error, __) -> DeleteGroupsRequest.getErrorResultCollection(groupList, error),
-                    log
-                ));
-
-            futures.add(future);
+            CompletableFuture<DeleteGroupsResponseData.DeletableGroupResultCollection> shareFuture = deleteShareGroups(topicPartition, groupList)
+                .thenCompose(groupErrMap -> {
+                    DeleteGroupsResponseData.DeletableGroupResultCollection collection = new DeleteGroupsResponseData.DeletableGroupResultCollection();
+                    List<String> errGroupIds = new ArrayList<>();
+                    groupErrMap.forEach((groupId, error) -> {
+                        if (error.code() != Errors.NONE.code()) {
+                            log.error("Error deleting share group {} due to error {}", groupId, error);
+                            errGroupIds.add(groupId);
+                            collection.add(
+                                new DeleteGroupsResponseData.DeletableGroupResult()
+                                    .setGroupId(groupId)
+                                    .setErrorCode(error.code())
+                            );
+                        }
+                    });
+
+                    Set<String> groupSet = new HashSet<>(groupList);
+                    // Remove all share group ids which have errored out
+                    // when deleting with persister.
+                    errGroupIds.forEach(groupSet::remove);
+
+                    // If no non-share groupIds or non-error share group ids present
+                    // return.
+                    if (groupSet.isEmpty()) {
+                        return CompletableFuture.completedFuture(collection);
+                    }
+
+                    // Let us invoke the standard procedure of any non-share
+                    // groups or successfully deleted share groups remaining.
+                    List<String> retainedGroupIds = groupSet.stream().toList();
+                    return runtime.scheduleWriteOperation(
```

Comment:
nit: It may be worth extracting this into a method to reduce the code in the `thenCompose`.